### PR TITLE
Minor: Improve comments on GenericByteViewArray::bytes_iter(), prefix_iter() and suffix_iter()

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Follow on to https://github.com/apache/arrow-rs/pull/6231 from @xinlifoobar 

# Rationale for this change
 
These new APIs were added in https://github.com/apache/arrow-rs/pull/6231 and I wanted to clarify their behavior (so we can potentailly use them in DataFusion)

# What changes are included in this PR?

Update doc comments

# Are there any user-facing changes?
Documentation, no functional change

### CODE REVIEW

File: arrow-array/src/array/byte_view_array.rs [hunk 1]
```diff
@@ -317,8 +317,11 @@ impl<T: ByteViewType + ?Sized> GenericByteViewArray<T> {
         })
     }
 
-    /// Returns an iterator over the prefix bytes of this array with respect to the prefix length.
-    /// If the prefix length is larger than the string length, it will return the empty slice.
+    /// Returns an iterator over the first `prefix_len` bytes of each array
+    /// element, including null values.
+    ///
+    /// If `prefix_len` is larger than the element's length, the iterator will
```
Comment (tustvold) [inline arrow-array/src/array/byte_view_array.rs:323]: This feels rather surprising to me?

Comment (alamb) [reply arrow-array/src/array/byte_view_array.rs:323]: Is that a statement or a question?

If you have a suggestion of how to make it less surprising perhaps @xinlifoobar would be willing to give it a try

Comment (tustvold) [reply arrow-array/src/array/byte_view_array.rs:323]: I guess I would have thought returning `Option<&[u8]>` would be perhaps more intuitive and likely perform similarly? That being said I am surprised that skipping the null mask yields performance returns, as this hasn't been my experience with regular StringArray, so my intuition about performance may be off

Comment (alamb) [reply arrow-array/src/array/byte_view_array.rs:323]: The results on https://github.com/apache/arrow-rs/pull/6312 seem to show that using null masks does indeed reduce performance

So is the conclusion that these APIs are good to go?

Comment (tustvold) [reply arrow-array/src/array/byte_view_array.rs:323]: I left some comments, I think this API would be better if it returned None for a prefix too long, even if it continues to not inspect null masks

**Review by tustvold — APPROVED** [review]
The docs look good, we can always update should we alter the signature

**Comment by alamb** [comment]
Ok, merging in these docs so they match the current code. As @tustvold  says if the code needs to be updated we can also update the docs

**Pull request merged**